tools/crm_resource.c

```diff
@@ -965,7 +965,70 @@ main(int argc, char **argv)
         rc = pcmk_ok;
 
     } else if (rsc_cmd == 'e') {
+        xmlXPathObject *xpathObj = NULL;
+        xmlNode *cib_constraints = NULL;
+        crm_time_t *now = crm_time_new(NULL);
+        int i;
+
+        cib_constraints = get_object_root(XML_CIB_TAG_CONSTRAINTS,
+                                          data_set->input);
+        xpathObj = xpath_search(cib_constraints, "//" XML_CONS_TAG_RSC_LOCATION);
+
+        for (i = 0; i < numXpathResults(xpathObj); i++) {
+            xmlNode *constraint_node = getXpathResult(xpathObj, i);
+            xmlNode *date_expr_node = NULL;
+            crm_time_t *end = NULL;
+            char *xpath_string = NULL;
+
+            if (!crm_starts_with(ID(constraint_node), "cli-ban-") ||
+                !crm_starts_with(ID(constraint_node), "cli-prefer-")) {
+                continue;
+            }
+
+            /* Find the date_expression within this constraint. */
+            xpath_string = crm_strdup_printf("//date_expression[@id='%s-lifetime']",
```

Comment:
cli-prefer-* constraints are named like 'cli-prefer-lifetime-end-RSC', which is horribly inconsistent, but what we have to work with.